File: backend/mission10_api/Models/Bowler.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace mission10_api.Models;

public partial class Bowler
{
    public int BowlerId { get; set; }

    public string? BowlerLastName { get; set; }

    public string? BowlerFirstName { get; set; }

    public string? BowlerMiddleInit { get; set; }

    public string? BowlerAddress { get; set; }

    public string? BowlerCity { get; set; }

    public string? BowlerState { get; set; }

    public string? BowlerZip { get; set; }

    public string? BowlerPhoneNumber { get; set; }
    
    [ForeignKey("TeamId")]
    public int? TeamId { get; set; }
    public virtual Team? Team { get; set; }
}

File: frontend/src/types/bowler.ts
export type bowler =  {
    "bowlerId": number,
    "bowlerLastName": string,
    "bowlerFirstName": string,
    "bowlerMiddleInit": string,
    "bowlerAddress": string,
    "bowlerCity": string,
    "bowlerState": string,
    "bowlerZip": number,
    "bowlerPhoneNumber": string,
    "teamId": number,
    "team": {
        "teamId": number,
        "teamName": string,
        "captainId": number
    }
}